addons/wp/functions/fnc_doArtillery.sqf

```diff
@@ -49,47 +50,50 @@ if !((vehicle _gun) isKindOf "StaticMortar") then {
 private _ammo = (getArtilleryAmmo [_gun]) param [0, ""];
 private _time = _gun getArtilleryETA [_center, _ammo];
 
-// delay
-private _mainStrike = linearConversion [100, 2000, (_gun distance2D _pos), 30, 90, true];
-private _checkRounds = (_time + random 35);
+// delay ~ no delay if skipping checkRounds nkenny
+private _mainStrike = [linearConversion [100, 2000, (_gun distance2D _pos), 30, 90, true], 0] select _skipCheckrounds;
+private _checkRounds = _time + random 35;
 
-// delay
+// delay for main strike
 sleep _mainStrike;
 
-// initate attack (gun and caller must be alive)
+// debug marker list
+private _markerList = [];
+
+// initate attack ~ gun and caller must be alive
 if (canFire _gun && {_caller call EFUNC(main,isAlive)}) then {
 
-    // debug marker list
-    private _mlist = [];
+    // caller marker
+    if (EGVAR(main,debug_functions)) then {
+        private _markerCaller = [_caller, ["Spotter (%1M)", round (_caller distance2D _center)], "Color5_FD_F", "mil_arrow"] call EFUNC(main,dotMarker);
+        _markerCaller setMarkerDir (_caller getDir _center);
+        _markerList pushBack _markerCaller;
+    };
 
     if !(_skipCheckrounds) then {
         // check rounds
         for "_check" from 1 to (1 + random 2) do {
 
-            // check rounds barrage
-            for "_i" from 1 to (1 + random 1) do {
-
-                // Randomize target location
-                private _target = _center getPos [(100 + (random _accuracy * 2)) / _check, _direction + 50 - random 100];
-
-                private _ammo = (getArtilleryAmmo [_gun]) param [0, ""];
-                if (_ammo isEqualTo "") exitWith {};
-
-                if (_target inRangeOfArtillery [[_gun], _ammo]) then {
-                    // Fire round
-                    _gun commandArtilleryFire [_target, _ammo, 1];
-
-                    // debug
-                    if (EGVAR(main,debug_functions)) then {
-                        private _m = [_target, ["%1 %3 (check round %2)", getText (configOf _gun >> "displayName"), _i, groupID (group _gun)], "Color4_FD_F", "hd_destroy"] call EFUNC(main,dotMarker);
-                        _mlist pushBack _m;
-                    };
-                    // waituntil
-                    waitUntil {unitReady _gun};
-                } else {
-                    if (EGVAR(main,debug_functions)) then {
-                        ["Error Target position is not reachable with artillery"] call EFUNC(main,debugLog);
-                    };
+            // randomize target location
+            private _target = _center getPos [(100 + (random _accuracy * 2)) / _check, _direction + 45 - random 90];
+
+            private _ammo = (getArtilleryAmmo [_gun]) param [0, ""];
+            if (_ammo isEqualTo "") exitWith {};
+
+            if (_target inRangeOfArtillery [[_gun], _ammo]) then {
+
+                // check rounds barrage
+                _gun commandArtilleryFire [_target, _ammo, 1 + random 2];
```

Comment:
This may create issues with ACE_CSW because commandArtilleryFire only fires maximum the amounts of rounds left in the magazine, and with ACE_CSW that is always limited to 1. I think we need to keep it looping manually as it was

Reply:
for loop re-implemented.